### PR TITLE
Adds a debounce to the update function

### PR DESCRIPTION
**Pull request by josephsweeney**
This prevents the update function from being called too frequently.
It may need to be tweaked, but 5ms debounce felt responsive and is
hopefully good enough for lower end devices.

@DrewMonroe you had mentioned the performance wasn't great on an older computer, let me know if this improves it at all. If it doesn't, I'll probably just remove it, and I'll actually have to write an efficient rendering layer.

### CODE REVIEW

**Comment by DrewMonroe** [comment]
Finally got around to testing this on an older machine. The behavior is still laggy, unfortunately. I'm inclined to say that we just scrap this, since it doesn't seem to provide benefit, and would just complicate the code base. Thoughts?

**Comment by josephsweeney** [comment]
Yeah, scrapping it will be fine. I'm guessing that it doesn't help because we just need to render faster.

So `update()` is called every time the visual needs to be updated, so potentially every mouse move on the canvas. The lag is coming from the fact that the time it takes to update is taking more time than the time between `mousemove` events. This means we're missing potential move events. So to fix this we'll need to add some optimizations to not do work if we don't need to.

In summary, we should scrap this and actually optimize the way we do rendering.
